x-pack/agent/pkg/agent/application/action_store.go

```diff
@@ -0,0 +1,164 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package application
+
+import (
+	"fmt"
+
+	yaml "gopkg.in/yaml.v2"
+
+	"github.com/elastic/beats/x-pack/agent/pkg/agent/storage"
+	"github.com/elastic/beats/x-pack/agent/pkg/core/logger"
+	"github.com/elastic/beats/x-pack/agent/pkg/fleetapi"
+)
+
+// actionStore receives multiples actions to persist to disk, the implementation of the store only
+// take care of action policy change every other action are discarded. The store will only keep the
+// last good action on disk, we assume that the action is added to the store after it was ACK with
+// Fleet. The store is not threadsafe.
+type actionStore struct {
+	log    *logger.Logger
+	store  store
+	dirty  bool
+	action action
+}
+
+func newActionStore(log *logger.Logger, file string) (*actionStore, error) {
+	store := storage.NewDiskStore(file)
```

Comment:
what about injecting storage. it will be easier to test this with some kind of inmemory test store

Reply:
+1 not sure why I didn't do it in this case.